### DISCUSSION TITLE
NimBLE tutorial posted

### DISCUSSION THREAD

**Discussion by frankcohen** [h2zero/NimBLE-Arduino#573] [General]
I posted a tutorial article on using Arduino and Bluetooth Low Energy (BLE) on ESP32 as server and client combined, using Apache NimBLE. I include my notes and example source code. Glad for your feedback and upvotes. https://www.reddit.com/r/arduino/comments/158bd4j/arduino_and_ble_on_esp32_as_server_and_client/

**Comment by h2zero** [comment]
Hello Frank, sorry I haven't had time to comment on this. Good write-up and I appreciate the critique.  The part about the strings has been someone alleviated, setValue will take an Arduino string input.

**Reply by frankcohen** [reply]
Thanks, I am glad to help with NimBLE. It's a good piece of work, I appreciate your effort. I'm neck deep in code for my project (https://github.com/frankcohen/ReflectionsOS). I'm thinking I will contribute an abstraction and utility layer for NimBLE that makes it easier to get started. Also, expect another article from me on how I use NimBLE and compass/magnetometer readings to triangulate headings between BLE devices. That's coming soon. -Frank

**Reply by frankcohen** [reply]
In the past I post to reddit /esp32. It's unavailable for the Reddit Blackout. Where else should I be posting my NimBLE articles? -Frank
